x-pack/src/test/java/org/logstash/xpack/test/RSpecTests.java

```diff
@@ -12,24 +12,57 @@
 import java.nio.file.Path;
 import java.nio.file.Paths;
 import java.util.Arrays;
+import java.util.List;
+
+import org.jruby.Ruby;
 import org.jruby.runtime.builtin.IRubyObject;
 import org.junit.Assert;
 import org.junit.Test;
+
+import org.logstash.Logstash;
 import org.logstash.RubyUtil;
 import org.logstash.Rubyfier;
 
-public final class RSpecTests {
+public class RSpecTests {
+
+    static final Path LS_HOME;
+
+    static {
+        Path root;
+        if (System.getProperty("logstash.root.dir") == null) {
+            // make sure we work from IDE as well as when run with Gradle
+            root = Paths.get("").toAbsolutePath(); // LS_HOME/x-pack
+            if (root.endsWith("x-pack")) root = root.getParent();
+        } else {
+            root = Paths.get(System.getProperty("logstash.root.dir")).toAbsolutePath();
+        }
+
+        if (!root.resolve("versions.yml").toFile().exists()) {
+            throw new AssertionError("versions.yml not found is logstash root dir: " + root);
```

Comment:
Is there an action we'd like the developer to take if this error is encountered?

Reply:
so we previously did not care about what CWD is, the assert is there to fail early. 
it's really an assert the expectation is a Java test is run from an IDE -> cwd is `logstash`, or from gradle in which case `logstash.root.dir` property is properly set. this should be explanatory by following the `AssertionError` all the way up here (and looking at the code above) - :thinking: if we need to explain all of this in the error message here ...